### PR TITLE
key package id extensions

### PR DESCRIPTION
**Pull request by franziskuskiefer**
This implements support for the key package id extension and tests for it.
This works towards #32, #101 .

### CODE REVIEW

File: src/creds.rs [hunk 1]
```diff
@@ -173,16 +173,16 @@ impl Codec for BasicCredential {
...
+    fn decode(cursor: &mut Cursor) -> Result<Self, CodecError> {
+        let identity = decode_vec(VecSize::VecU16, cursor)?;
+        let ciphersuite = Ciphersuite::decode(cursor)?;
+        let public_key = SignaturePublicKey::decode(cursor)?;
+        Ok(BasicCredential {
+            identity,
+            ciphersuite,
+            public_key,
+        })
+    }
```
Comment (raphaelrobert) [inline src/creds.rs:185]: Would be nice to have encoding/decoding tests for both `Credential` and `BasicCredential`

**Review by raphaelrobert — CHANGES_REQUESTED** [review]

File: src/creds.rs (hunk 1 above)
Comment (franziskuskiefer) [reply src/creds.rs:185]: I agree, I filed #130 to write tests for encoding and decoding in general. This specific case might also get covered before when we look at credentials.

File: src/key_packages/mod.rs [hunk 2]
```diff
@@ -123,6 +137,15 @@ impl KeyPackage {
         None
     }
 
+    /// Get the ID of this key package as byte slice.
+    /// Returns an error if no Key ID extension is present.
+    pub fn get_id<'a>(&'a self) -> Result<&'a [u8], KeyPackageError> {
```
Comment (raphaelrobert) [inline src/key_packages/mod.rs:142]: Resolve clippy warning

File: src/extensions/key_package_id_extension.rs [hunk 3]
```diff
@@ -29,6 +29,11 @@ impl KeyIDExtension {
             key_id: id.to_vec(),
         }
     }
+
+    /// Get the value of the key id as byte slice.
+    pub(crate) fn as_slice<'a>(&'a self) -> &'a [u8] {
```
Comment (raphaelrobert) [inline src/extensions/key_package_id_extension.rs:34]: Resolve clippy warning

**Review by raphaelrobert — CHANGES_REQUESTED** [review]

**Review by raphaelrobert — APPROVED** [review]

**Pull request merged**